bpf/lib/host_firewall.h

```diff
@@ -284,15 +263,10 @@ ipv4_host_policy_egress(struct __ctx_buff *ctx, __u32 src_sec_identity,
 	__u16 node_id = 0;
 	__u16 proxy_port = 0;
 
-	if (src_sec_identity != HOST_ID) {
-#  ifndef ENABLE_MASQUERADE
-		return whitelist_snated_egress_connections(ctx, ipcache_srcid,
-							   trace, ext_err);
-#  else
-		/* Only enforce host policies for packets from host IPs. */
+	/* Only enforce host policies for packets from host IPs. */
+	if (src_sec_identity != HOST_ID &&
+	    (is_defined(ENABLE_MASQUERADE) || ipcache_srcid != HOST_ID))
```

Comment:
The comment doesn't match the condition anymore. Maybe we need to move the comment in `whitelist_snated_egress_connections` here?